source/agora/common/Hash.d

```diff
@@ -164,20 +164,22 @@ public void hashPart (ulong record, scope HashDg state) /*pure*/ nothrow @nogc @
 /// Ditto
 public void hashPart (in char[] record, scope HashDg state) /*pure*/ nothrow @nogc @trusted
 {
-    state(cast(const ubyte[])record);
+    hashPart(cast(const ubyte[])record, state);
 }
 
 /// Ditto
 public void hashPart (in ubyte[] record, scope HashDg state)
     /*pure*/ nothrow @nogc @safe
 {
+    hashPart(record.length, state);
```

Comment:
I think the return type of 'length' if platform dependent,so this might need to be explicitely converted to ulong like 
```D
hashPart(cast(ulong)record.length, state);
```

Reply:
Side note: If we use the serialization code, it takes care of this.